utils/log4cplus/log_utils.cpp

```diff
@@ -0,0 +1,50 @@
+#include "log_utils.h"
+ 
+#include <log4cplus/logger.h>
+#include <log4cplus/layout.h>
+#include <log4cplus/configurator.h>
+ 
+Logger logger = Logger::getInstance(LOG4CPLUS_TEXT("logmain"));
+ 
+void initLogger(bool isDebug) {
+    if (isDebug) {
+       PropertyConfigurator::doConfigure(LOG4CPLUS_TEXT("./log4cplus_d.conf"));
+    } else {
+    	PropertyConfigurator::doConfigure(LOG4CPLUS_TEXT("./log4cplus.conf"));
+    }
+}
+
+#define DO_LOGGER(logLevel, pFormat, bufSize)\
+    if(logger.isEnabledFor(logLevel))\
+{                \
+    va_list args;            \
+    va_start(args, pFormat);        \
+    char buf[bufSize] = {0};        \
+    VSPRINTF(buf, sizeof(buf), pFormat, args);    \
+    va_end(args);           \
+    logger.forcedLog(logLevel, buf); \
+}
+
+void log_d(const char* pFormat,  ...)
+{
+    DO_LOGGER(log4cplus::DEBUG_LOG_LEVEL, pFormat, 1024);
+}
+
+void log_e(const char* pFormat,  ...)
+{
+    DO_LOGGER(log4cplus::ERROR_LOG_LEVEL, pFormat, 1024);
+}
+
+void log_i(const char* pFormat,  ...)
+{
+    DO_LOGGER(log4cplus::INFO_LOG_LEVEL, pFormat, 1024);
+}
+
+void log_w(const char* pFormat,  ...)
+{
+    DO_LOGGER(log4cplus::WARN_LOG_LEVEL, pFormat, 1024);
+}
+
+void shutDown(){
```

Comment:
rename to loggerShutdown or something else.

Reply:
done.